From 8e4a863b099c9dbb650cbd48e7e966a45be193f5 Mon Sep 17 00:00:00 2001
From: Liu Zhiquan <zhiquan.liu@intel.com>
Date: Thu, 6 Jun 2019 16:05:04 +0800
Subject: [PATCH 2/3] [RVERTME]write '0' as empty InputWindowInfo to align the
 data

The ligbui data between the system and vndk is not alignment.
The data transfer from vndk libgui through binder is without
InputWindowInfo, but system libgui will read InputWindowInfo.
in order to aligned them, wirte "0" to data, system libgui will
tread it as an empty InputWindowInfo.

Change-Id: I364b1c6be54bf8549868ce6dfbd7194d5487cf82
Tracked-On:
Signed-off-by: Liu Zhiquan <zhiquan.liu@intel.com>
---
 libs/gui/LayerState.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libs/gui/LayerState.cpp b/libs/gui/LayerState.cpp
index 6066421fa..26b68b98a 100644
--- a/libs/gui/LayerState.cpp
+++ b/libs/gui/LayerState.cpp
@@ -54,6 +54,8 @@ status_t layer_state_t::write(Parcel& output) const
     output.writeFloat(color.b);
 #ifndef NO_INPUT
     inputInfo.write(output);
+#else
+    output.writeInt32(0);
 #endif
     output.write(transparentRegion);
     output.writeUint32(transform);
--
2.21.0
